### PR TITLE
task 8 completed

### PR DESCRIPTION
**Pull request by SardarAmiri**
task 8 completed

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -241,4 +241,57 @@ describe('Integreation App Testing For EndPoints', () => {
             
         })
     })
+    describe('CORE: PATCH /api/articles/:article_id' , () => {
+        test('status 200: Responds with the updated article', () => {
+            const updatedVote = {inc_votes: 10}
```
Comment (nc-mitch-wyatt) [inline __tests__/app.test.js:246]: This test is brilliant 🏆

File: __tests__/app.test.js [hunk 2]
```diff
@@ -241,4 +241,57 @@ describe('Integreation App Testing For EndPoints', () => {
...
+                        author: "butter_bridge",
+                        body: "I find this existence challenging",
+                        votes: 110,
+                        article_img_url:
+                        "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700"
+                      }
+                })
+            })
+        })
+        test('400 status: Responde Bad request when patching with invalid id ', () => {
```
Comment (nc-mitch-wyatt) [inline __tests__/app.test.js:266]: respondé 🤌 🇪🇸

File: __tests__/app.test.js [hunk 3]
```diff
@@ -241,4 +241,57 @@ describe('Integreation App Testing For EndPoints', () => {
...
+            const updatedVote = {inc_votes: 10}
+            return request(app)
+            .patch('/api/articles/invalid_id')
+            .send(updatedVote)
+            .expect(400)
+            .then((response) => {
+                expect(response.body.msg).toBe('Bad request')
+            })
+        })
+        test('404 status: Responde No Content when update an article ', () => {
```
Comment (nc-mitch-wyatt) [inline __tests__/app.test.js:276]: *an article that doesn't exist

File: __tests__/app.test.js [hunk 4]
```diff
@@ -241,4 +241,57 @@ describe('Integreation App Testing For EndPoints', () => {
...
+            })
+        })
+        test('400 status: Responde Bad request when patching with invalid id ', () => {
+            const updatedVote = {inc_votes: 10}
+            return request(app)
+            .patch('/api/articles/invalid_id')
+            .send(updatedVote)
+            .expect(400)
+            .then((response) => {
+                expect(response.body.msg).toBe('Bad request')
```
Comment (nc-mitch-wyatt) [inline __tests__/app.test.js:293]: This test is a duplicate ⬆️ 
What about a test for if the _inc_votes_ is not a number ?

File: nc-news-model/nc-news.model.js [hunk 5]
```diff
@@ -56,6 +56,17 @@ module.exports.createCommentOnArticle = (article_id, {username, body}) => {
...
     })
-    
+}
+
+module.exports.updateArticleVoteById = (article_id, inc_votes) => {
+
+    return db.query(`UPDATE articles SET votes = votes + $1 WHERE article_id = $2 RETURNING *`, [inc_votes, article_id])
+    .then((result) => {
+        if(result.rows.length === 0){
+            const err = new CustomError(`No user found for article_id ${article_id}`, 404)
```
Comment (nc-mitch-wyatt) [inline nc-news-model/nc-news.model.js:66]: This error message is good but remember it's not that a _user_ hasn't been found, it's that _no article has been found_

**Review by nc-mitch-wyatt — COMMENTED** [review]
Missing a test for if _inc_votes_ is not a number, and some other small things in comments below ⬇️

**Comment by SardarAmiri** [comment]
I'll refactor these stuff along task 9

**Pull request merged**